src/hands/KM_HandSpectator.pas

```diff
@@ -172,6 +209,18 @@ procedure TKMSpectator.UpdateSelect;
   //Don't clear the old selection unless we found something new
   if NewSelected <> nil then
     Selected := NewSelected;
+
+  // In a replay we want in-game statistics (and other things) to be shown for the owner of the last select object
+  if gGame.GameMode in [gmMultiSpectate, gmReplaySingle, gmReplayMulti] then
+  begin
+    UID := UID_NONE;
+    if Selected is TKMHouse      then begin HandIndex := TKMHouse    (Selected).Owner; UID := TKMHouse(Selected).UID; end;
+    if Selected is TKMUnit       then begin HandIndex := TKMUnit     (Selected).Owner; UID := TKMHouse(Selected).UID; end;
+    if Selected is TKMUnitGroup  then begin HandIndex := TKMUnitGroup(Selected).Owner; UID := TKMHouse(Selected).UID; end;
```

Comment:
You forgot to change "TKMHouse" for the second two. Probably best to expand it rather than stacking several statements in one line, so mistakes like this are easier to spot